appbuilder/project/project-base.ts

```diff
@@ -80,6 +85,56 @@ export abstract class ProjectBase implements Project.IProjectBase {
 		};
 	}
 
+	public getAppIdentifierForPlatform(platform?: string): IFuture<string> {
+		return ((): string => {
+			if (!this._platformSpecificAppIdentifier) {
+				if (platform &&
+					platform.toLowerCase() === this.$projectConstants.ANDROID_PLATFORM_NAME.toLowerCase() &&
+					this.projectData.Framework === TARGET_FRAMEWORK_IDENTIFIERS.Cordova) {
+					let pathToAndroidResources = path.join(this.projectDir, this.$staticConfig.APP_RESOURCES_DIR_NAME, this.$projectConstants.ANDROID_PLATFORM_NAME);
+
+					let pathToAndroidManifest = path.join(pathToAndroidResources, ProjectBase.ANDROID_MANIFEST_NAME);
+					let appIdentifierInAndroidManifest = this.getAppIdentifierFromConfigFile(pathToAndroidManifest, /package\s*="([\S]*)"/).wait();
+
+					let pathToConfigXml = path.join(pathToAndroidResources, "xml", ProjectBase.CONFIG_XML_NAME);
+
+					let appIdentifierInConfigXml = this.getAppIdentifierFromConfigFile(pathToConfigXml, /id\s*="([\S]*)"/).wait();
+
+					if (appIdentifierInAndroidManifest && appIdentifierInConfigXml && appIdentifierInAndroidManifest === appIdentifierInConfigXml) {
+						if (appIdentifierInAndroidManifest === ProjectBase.APP_IDENTIFIER_PLACEHOLDER) {
+							this._platformSpecificAppIdentifier = this.projectData.AppIdentifier;
+						} else {
+							this._platformSpecificAppIdentifier = appIdentifierInAndroidManifest;
+						}
+					} else if (appIdentifierInAndroidManifest || appIdentifierInConfigXml) {
+						this.$errors.failWithoutHelp(this.getInvalidAppIdentifierMessage());
+					}
+				} else {
+					// Since we don't have specific logic to get the app identifier for iOS and WP8 we can return the one from .abproject file.
+					this._platformSpecificAppIdentifier = this.projectData.AppIdentifier;
+				}
+			}
+
+			return this._platformSpecificAppIdentifier;
+		}).future<string>()();
+	}
+
+	public isAppIdentifierValid(platform: string): IFuture<boolean> {
+		return ((): boolean => {
+			try {
+				this.getAppIdentifierForPlatform(platform).wait();
```

Comment:
You can shorten this method a bit by using a return statement here like:
```TypeScript
return !!this.getAppIdentifierForPlatform(platform).wait();
```
and skipping the `return` statement after the `try ... catch ... ` clause.
Just a minor suggestion and definitely not a merge-stopper